.github/workflows/sanitize.yml

```diff
@@ -7,124 +7,47 @@ on:
+        name:      ["Sanitize"]
+        sanitizer: ["address", "undefined", "thread"]
         simd:      ["ON", "OFF"]
-        sanitizer: ["address", "leak", "undefined", "memory", "thread"]
     env:
-        CC:         ${{ matrix.cc }}
-        CXX:        ${{ matrix.cxx }}
-        # We set PYTHONPATH instead of installing arbor to avoid distribution/OS specific behaviour.
-        PYTHONPATH: ${{ github.workspace }}/build/python
+        CC:           clang-10
+        CXX:          clang++-10
+        ASAN_OPTIONS: detect_leaks=1
     steps:
       - name: Set up cmake
         uses: jwlawson/actions-setup-cmake@v1.7
         with:
-          cmake-version: ${{ matrix.cmake }}
-      - name: Set up Python
-        uses: actions/setup-python@v2
-        with:
-          python-version: ${{ matrix.py }}
-      - name: OpenMPI cache
-        uses: actions/cache@v2
-        id:   cache-ompi
-        with:
-          path: ~/openmpi-4.0.2
-          key:  ${{ matrix.os }}-openmpi-4.0.2-${{ matrix.cxx }}
-      - name: Build OpenMPI
-        if: ${{ steps.cache-ompi.outputs.cache-hit != 'true' }}
-        run: |
-           echo cache-hit='${{ steps.cache-ompi.outputs.cache-hit }}'
-           cd ~
-           wget https://download.open-mpi.org/release/open-mpi/v4.0/openmpi-4.0.2.tar.gz
-           tar -xvf ./openmpi-4.0.2.tar.gz
-           cd openmpi-4.0.2
-           ./configure --disable-mpi-fortran
-           make -j4
-      - name: Install OpenMPI
-        run: |
-           echo "Going to install ompi"
-           cd ~
-           cd openmpi-4.0.2
-           sudo make install
-           cd -
-      - name: Update shared library cache
-        if: ${{ startsWith(matrix.os, 'ubuntu') }}
-        run: sudo ldconfig
-      - name: Install Python packages
-        run:  pip install numpy sphinx svgwrite sphinx-rtd-theme mpi4py pandas seaborn
+          cmake-version: 3.19.x
       - name: Clone w/ submodules
         uses: actions/checkout@v2
         with:
           submodules: recursive
-      - name: Check config
-        run: |
-          $CC --version
-          $CXX --version
-          python --version
-          mpic++ --show
-          mpicc --show
-          echo $PYTHONPATH
       - name: Build arbor
         run: |
           mkdir build
           cd build
-          cmake .. -DCMAKE_BUILD_TYPE=debug -DCMAKE_CXX_FLAGS="-fsanitize=${{ matrix.sanitizer }} -fno-omit-frame-pointer" -DCMAKE_C_FLAGS="-fsanitize=${{ matrix.sanitizer }} -fno-omit-frame-pointer" -DCMAKE_EXE_LINKER_FLAGS="-fsanitize=${{ matrix.sanitizer }}" -DCMAKE_MODULE/_LINKER_FLAGS="-fsanitize=${{ matrix.sanitizer }}" -DCMAKE_CXX_COMPILER=$CXX -DCMAKE_C_COMPILER=$CC -DARB_WITH_PYTHON=ON -DARB_VECTORIZE=${{ matrix.simd }} -DPython3_EXECUTABLE=`which python` -DARB_WITH_MPI=${{ matrix.mpi }} -DARB_USE_BUNDLED_LIBS=ON
-          make -j4 tests examples pyarb html
+          export LD_PRELOAD=$(clang -print-file-name=libclang_rt.san-x86_64.so)
```

Comment:
The preload is generating a bunch of warnings in the build log; can be removed (pers. comm. with @thorstenhater).